ang/civicase/ContactCard.js

```diff
@@ -0,0 +1,126 @@
+(function (angular, $, _, CRM) {
+  var module = angular.module('civicase');
+
+  module.directive('contactCard', function ($document, ContactsDataService) {
+    return {
+      restrict: 'A',
+      templateUrl: '~/civicase/ContactCard.html',
+      scope: {
+        data: '=contacts',
+        isAvatar: '=avatar'
+      },
+      link: contactCardLink
+    };
+
+    function contactCardLink (scope, elem, attrs) {
+      scope.getContactIconOf = ContactsDataService.getContactIconOf;
+      scope.isPopupVisible = false;
+      scope.ts = CRM.ts('civicase');
+      scope.url = CRM.url;
+
+      (function init () {
+        scope.$watch('data', refresh);
+        $document.bind('click', closeDropdownIfClickedOutside);
+      }());
+
+      /**
+       * Toggle dropdown visibility
+       *
+       * @param {object} $event
+       */
+      scope.toggleDropdownVisibility = function ($event) {
+        scope.isPopupVisible = !scope.isPopupVisible;
+        $event.stopPropagation();
+      };
+
+      /**
+       * Close the Dropdown when clicked outside
+       *
+       * @param {object} event
+       */
+      function closeDropdownIfClickedOutside (event) {
+        var isClickedElementChildOfPopup = elem
+          .find(event.target)
+          .length > 0;
+
+        if (isClickedElementChildOfPopup) {
+          return;
+        }
+
+        scope.isPopupVisible = false;
+        scope.$apply();
```

Comment:
We are reinitiating the dirty digest cycle. Is the use of `scope.$apply` necessary ?

Reply:
Yes because `closeDropdownIfClickedOutside` is fired outside of angular context, so need to call `scope.$apply` to initiate digest cycle.